### ISSUE TITLE
static methods on RequestFactory Proxies

### ISSUE THREAD

**Issue by dodgex** [gwtproject/gwt#9291]
similar to https://github.com/gwtproject/gwt/issues/9289 but for static methods

it would be nice to have static methods on proxies used for requestfactory

``` java
@ProxyFor(value = StatsAccount.class, locator = HibernateLocator.class)
public interface StatsAccountProxy extends EntityProxy {

     // setter & getter

    static StatsAccountProxy create(RequestContext request,
            TenantProxy tenant) {
        TenantProxy editTenant = request.edit(tenant);
        StatsAccountProxy entity = request.create(StatsAccountProxy.class);
        entity.setQueries(new ArrayList<StatsQueryProxy>());
        entity.setTrees(new ArrayList<TreeProxy>());
        entity.setTenant(editTenant);
        return entity;
    }
}
```

**Comment by niloc132** [comment]
Initial patch to make these details available to generators so it knows that implementors of the interface are not required to implement this: https://gwt-review.googlesource.com/#/c/14750

More work may be required, this patch is more addressed toward #9289.

**Comment by niloc132** [comment]
Closing, Generators will not support java8+ features (as discussed in the linked CL). Instead, any generated code should be updated to annotation processors instead. 

Interestingly, RequestFactory actually already uses annotation processors, but only for validating at runtime that the expected pieces are present to work properly - at the same time, it is the only annotation processor which hasn't had any interest shown in the community to be updated to annotation processors, and nearly all of those have been completely ported.
